### PR TITLE
Move Player creation out of PlayerManager

### PR DESCRIPTION
**Pull request by defunctzombie**
**User-Facing Changes**
None

**Description**
This change moves player creation out of PlayerManager and into individual factory instances. Each factory is responsible for the initialization of players of that specific type.

This change purposefully does not change the UX around selecting data sources. This change keeps file/directory dialog and prompt control within PlayerManager but changes the logic to use settings/hints from the selected data source factory.

Fixes: https://github.com/foxglove/studio/issues/776

### CODE REVIEW

**Comment by jtbandes** [comment]
🙏

File: packages/studio-base/src/playerFactories/Ros2SocketPlayerFactory.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import { Player } from "@foxglove/studio-base/players/types";
+
+class Ros2SocketPlayerFactory implements IPlayerFactory {
+  id = "ros2-socket";
+  displayName = "ROS 2";
+  iconName: IPlayerFactory["iconName"] = "studio.ROS";
+
+  promptOptions(previousValue?: string): PromptOptions {
+    return {
+      title: "ROS 2 DomainId",
```
[the remark this answers is not in the document]
Comment (jtbandes) [reply packages/studio-base/src/playerFactories/Ros2SocketPlayerFactory.tsx:20]: Google also knows the answer: https://docs.ros.org/en/foxy/Concepts/About-Domain-ID.html

[the remark this answers is not in the document]
Comment (jhurliman) [reply packages/studio-base/src/playerFactories/Ros2SocketPlayerFactory.tsx:20]: `ROS_DOMAIN_ID` is the env var

File: packages/studio-base/src/components/PlayerManager.tsx [hunk 2]
```diff
@@ -43,376 +42,53 @@ import { usePrompt } from "@foxglove/studio-base/hooks/usePrompt";
...
-}
-
-export default function PlayerManager({
-  children,
-  playerSources,
-}: PropsWithChildren<{
-  playerSources: PlayerSourceDefinition[];
-}>): JSX.Element {
-  useWarnImmediateReRender();
+  // When we implement per-data-connector UI settings we will move this into the foxglove data platform source.
```
[the remark this answers is not in the document]
Comment (defunctzombie) [reply packages/studio-base/src/components/PlayerManager.tsx:89]: We'll pretend it didn't happen.

**Pull request merged**